crates/stackable-versioned-macros/src/lib.rs

```diff
@@ -8,50 +8,95 @@ mod attrs;
 mod codegen;
 mod consts;
 
-/// This macro enables generating versioned structs.
+/// This macro enables generating versioned structs and enums.
 ///
-/// ## Usage Guide
+/// # Usage Guide
 ///
-/// ### Quickstart
+/// In this guide, code blocks usually come in pairs. The first code block
+/// describes how the macro is used. The second expandable block displays the
+/// generated piece of code for explanation purposes. It should be noted, that
+/// the exact code can diverge from what is being depicted in this guide. For
+/// example, `#[automatically_derived]` and `#[allow(deprecated)]` are removed
+/// in most examples to reduce visual clutter.
+///
+/// ## Declaring Versions
+///
+/// It is **important** to note that this macro must be placed before any other
+/// (derive) macros and attributes. This ensures that the macros and attributes
+/// are applied to the generated versioned instances of the struct or enum.
```

Comment:
```suggestion
/// It is **important** to note that this macro must be placed before any other
/// (derive) macros and attributes. Macros supplied before the versioned macro
/// will be erased, because the original struct or enum (container) is erased,
/// and new containers are generated. This ensures that the macros and
/// attributes are applied to the generated versioned instances of the
/// container.
```